src/brogue/RogueMain.c

```diff
@@ -744,12 +744,22 @@ void startLevel(short oldLevelNumber, short stairDirection) {
     px = player.xLoc;
     py = player.yLoc;
     player.xLoc = player.yLoc = 0;
-    for (i = 0; i < 100 && i < (short) timeAway; i++) {
+    timeAway = min(timeAway, 100);
+    while (timeAway--) {
+        unsigned long turn = rogue.absoluteTurnNumber;
+        rogue.absoluteTurnNumber = (turn > timeAway ? turn - timeAway : 0);
         updateEnvironment();
+        rogue.absoluteTurnNumber = turn;
     }
```

Comment:
Good solution! It stretches the concept of absoluteTurnNumber a little but it's quite natural. nit: I think having the store/load outside the loop might be a bit more readable? If you don't want to leak the var i guess it could go in a scope block too

```suggestion
    unsigned long currentTurn = rogue.absoluteTurnNumber;
    while (timeAway--) {
        rogue.absoluteTurnNumber = (currentTurn > timeAway ? currentTurn - timeAway : 0);
        updateEnvironment();
    }
    rogue.absoluteTurnNumber = currentTurn;
```

Reply:
Agreed!